### PR TITLE
iqiyi: support more stream quality

### PR DESCRIPTION
**Pull request by zhangn1985**
algorism form @ERioK
thank you @ERioK

Signed-off-by: Zhang Ning <zhangn1985@gmail.com>

for #1211 
but no M3u8 parsing..

```
python3 you-get -id http://www.iqiyi.com/v_19rrldw9sg.html 
[DEBUG] get_content: http://cache.m.iqiyi.com/tmts/493448700/a0326d0224db24415f2c0ccacc800266/?t=1467202675054&sc=99cf652172f294d864d967ec6ce7d8ba&src=76f90cbd92f94a2e925d83e8ccd22cb7
site:                爱奇艺 (Iqiyi)
title:               狭路第1集
streams:             # Available quality and codecs
    [ DEFAULT ] _________________________________
    - format:        BD
      container:     m3u8
      video-profile: 全高清
      size:          0.0 MiB (0 bytes)
    # download-with: you-get --format=BD [URL]

    - format:        FD
      container:     m3u8
      video-profile: 超高清
      size:          0.0 MiB (0 bytes)
    # download-with: you-get --format=FD [URL]

    - format:        TD
      container:     m3u8
      video-profile: 超清
      size:          0.0 MiB (0 bytes)
    # download-with: you-get --format=TD [URL]

    - format:        HD
      container:     m3u8
      video-profile: 高清
      size:          0.0 MiB (0 bytes)
    # download-with: you-get --format=HD [URL]

    - format:        SD
      container:     m3u8
      video-profile: 标清
      size:          0.0 MiB (0 bytes)
    # download-with: you-get --format=SD [URL]

    - format:        LD
      container:     m3u8
      video-profile: 流畅
      size:          0.0 MiB (0 bytes)
    # download-with: you-get --format=LD [URL]
```

---
This change is [Reviewable](https://reviewable.io/reviews/soimort/you-get/1236)

### CODE REVIEW

**Comment by ERioK** [comment]
@zhangn1985  
清晰度列表好像有点问题，漏了1080p，1080p的vid是在ctl段里面的，需要用这个vid再请求一次json可以得到地址
不知道能不能解析这个m3u8，里面有每个分段的地址，这样就可以全部下载下来合并了

**Comment by goodneo** [comment]
@ERioK 
http://cache.video.qiyi.com/vms?key=fvip&src=

想請問一下從這裡面看到了vid沒辦法抓出來嗎?

**Comment by ERioK** [comment]
@goodneo  只要有vid和tvid，用那个算法得到url，data.m3utx 里面就是播放的地址了

File: src/you_get/extractors/iqiyi.py [hunk 1]
```diff
@@ -11,8 +11,6 @@
 
 import time
 
-from .iqiyi_sc import gen_sc
```
Comment (soimort) [inline src/you_get/extractors/iqiyi.py]: So now you can `git rm iqiyi_sc.py`

File: src/you_get/extractors/iqiyi.py [hunk 2]
```diff
@@ -86,26 +84,31 @@ class Iqiyi(VideoExtractor):
...
-
+    '''
     supported_stream_types = [ 'high', 'standard']
 
 
     stream_to_bid = {  '4k': 10, 'fullhd' : 5, 'suprt-high' : 4, 'super' : 3, 'high' : 2, 'standard' :1, 'topspeed' :96}
+    '''
+    ids = ['BD', 'FD', 'OD', 'TD', 'HD', 'SD', 'LD']
+    vd_2_id = {21: 'TD', 2: 'HD', 4: 'FD', 17: 'BD', 96: 'LD', 1: 'SD'}
+    vd_2_profile = {21: u'超清', 2: u'高清', 4: u'超高清', 17: u'全高清', 96: u'流畅', 1: u'标清'}
```
Comment (soimort) [inline src/you_get/extractors/iqiyi.py]: The Unicode literal syntax isn't available until 3.3. Just drop it.

**Comment by goodneo** [comment]
@ERioK 有看到你說的ctl段 但不知道怎麽組合出m3u8 url

**Comment by soimort** [comment]
FFmpeg can dump an M3U8 URL to a local file

```
$ ffmpeg -i http://live.schou.me/fish.m3u8 -t 20 -c copy -bsf:a aac_adtstoasc output.mp4
```

...but certainly it's my responsibility to implement that common part.

**Comment by cnbeining** [comment]
It would be great to have a working FFMpeg to deal with M3U. We could use this to work on HLS...

**Comment by zhangn1985** [comment]
@ERioK info['data']['ctl']的确是有h264_1080p的vid.

这个先放一边。

info['data']['m3utx'] 和 info['data']['m3u'] 分别是什么呢？

**Comment by ERioK** [comment]
@zhangn1985  我反正没看出区别来，随便用一个吧

**Comment by zhangn1985** [comment]
所有关心的同学：
```
vd_2_id = {
        96: '210P-H264',
        1:  '360P-H264',
        2:  '540P-H264',
        21: '540P-H265',
        4:  '720P-H264',
        17: '720P-H265',
        5:  '1080P-H264',
        18: '1080P-H265',
    }
```
那么多stream profile，是要一一列出来吗？还是按size 合并吗？

**Comment by goodneo** [comment]
@zhangn1985 感覺好像按size比較好  差別在?

**Comment by zhangn1985** [comment]
因为有8个profile，5个size，按size合并看上去好看一些

**Comment by goodneo** [comment]
@zhangn1985 看上面還有分H264 H265 那就照你說的吧~~

**Comment by wwqgtxx** [comment]
我还遇到过14这个清晰度，貌似也是720P的h264，也是醉了

**Comment by zhangn1985** [comment]
电脑太老了，解码H265费力的要死。

**Comment by wwqgtxx** [comment]
@zhangn1985 i7 4790表示无压力

**Comment by goodneo** [comment]
@zhangn1985 想請問一下要怎麼把m3u8轉mp4?

**Comment by cnbeining** [comment]
要么获取所有的link 下载 合并

要么用我的那个pr 直接合并。

2016年6月30日星期四，goodneo <notifications@github.com> 写道：

**Comment by goodneo** [comment]
@cnbeining 有載了github但不懂怎麼使用..

**Comment by soimort** [comment]
@cnbeining I thought you've made ffmpeg handle the M3U8 from iQIYI , but why am I still getting a raw M3U8 file?

```
[♥] you-get  develop  ./you-get -d "http://www.iqiyi.com/v_19rrli054g.html#vfrm=8-8-0-1"
[DEBUG] get_content: http://cache.m.iqiyi.com/tmts/502278300/b3336c7d61bd0585da883066f3ca978e/?t=1467380541199&sc=cef4d843c1775b7bd5bc768e2446820a&src=76f90cbd92f94a2e925d83e8ccd22cb7
site:                爱奇艺 (Iqiyi)
title:               2016温网自制HL&nbsp;&nbsp;科贝尔VS罗布森 自制HL
stream:
    - format:        TD
      container:     m3u8
      video-profile: 720p
      size:          0.0 MiB (0 bytes)
    # download-with: you-get --format=TD [URL]

Downloading 2016温网自制HL  科贝尔VS罗布森 自制HL.m3u8 ...
 100% (  0.0/  0.0MB) ├███████████████████████████████████████████┤[1/1]    2 kB/s
```

**Comment by soimort** [comment]
Oh, forget what I just said. iQIYI was implemented in `VideoExtractor`, which makes it harder to bypass the prescribed downloading workflow.

**Comment by charlie20083** [comment]
So, downloading a video on iQiYi will only get a .m3u8 file instead of a video file is excepted in this version or not?

**Comment by cnbeining** [comment]
My fault.

Anyone can get a quick fix refering Showroom before I can grab a machine?

2016年7月1日星期五，charlie20083 <notifications@github.com> 写道：

**Comment by charlie20083** [comment]
So if we find someone can handle that .m3u8 file and we will have the correct video file maybe?
but I'm not familiar with Python...

**Comment by cnbeining** [comment]
Actually we DO have the method.

I am just thinking whether it is worth it to use ffmpeg since this is not
live....

2016年7月1日星期五，charlie20083 <notifications@github.com> 写道：

**Comment by gdgsdg123** [comment]
> Quote [zhangn1985](https://github.com/soimort/you-get/pull/1236#issue-75637926):
>
> ```ShellSession
> > python3 you-get -d -i "http://www.iqiyi.com/v_19rrldw9sg.html"
>
> [DEBUG] get_content: http://cache.m.iqiyi.com/tmts/493448700/a0326d0224db24415f2c0ccacc800266/?t=1467202675054&sc=99cf652172f294d864d967ec6ce7d8ba&src=76f90cbd92f94a2e925d83e8ccd22cb7
> site:                爱奇艺 (Iqiyi)
> title:               狭路第1集
> streams:             # Available quality and codecs
>     [ DEFAULT ] _________________________________
>     - format:        BD
>       container:     m3u8
>       video-profile: 全高清
>       size:          0.0 MiB (0 bytes)
>     # download-with: you-get --format=BD [URL]
>
>     - format:        FD
>       container:     m3u8
>       video-profile: 超高清
>       size:          0.0 MiB (0 bytes)
>     # download-with: you-get --format=FD [URL]
>
>     - format:        TD
>       container:     m3u8
>       video-profile: 超清
>       size:          0.0 MiB (0 bytes)
>     # download-with: you-get --format=TD [URL]
>
>     - format:        HD
>       container:     m3u8
>       video-profile: 高清
>       size:          0.0 MiB (0 bytes)
>     # download-with: you-get --format=HD [URL]
>
>     - format:        SD
>       container:     m3u8
>       video-profile: 标清
>       size:          0.0 MiB (0 bytes)
>     # download-with: you-get --format=SD [URL]
>
>     - format:        LD
>       container:     m3u8
>       video-profile: 流畅
>       size:          0.0 MiB (0 bytes)
>     # download-with: you-get --format=LD [URL]
> ```

> Quote "[/src/you_get/extractors/iqiyi.py](https://github.com/soimort/you-get/blob/398068e13e8b7052903f89d1caf30023e8639e78/src/you_get/extractors/iqiyi.py)":
>
> ```Python
> class Iqiyi(VideoExtractor):
>     name = "爱奇艺 (Iqiyi)"
>
>     stream_types = [
>         {'id': '4k', 'container': 'm3u8', 'video_profile': '4k'},
>         {'id': 'BD', 'container': 'm3u8', 'video_profile': '1080p'},
>         {'id': 'TD', 'container': 'm3u8', 'video_profile': '720p'},
>         {'id': 'TD_H265', 'container': 'm3u8', 'video_profile': '720p H265'},
>         {'id': 'HD', 'container': 'm3u8', 'video_profile': '540p'},
>         {'id': 'HD_H265', 'container': 'm3u8', 'video_profile': '540p H265'},
>         {'id': 'SD', 'container': 'm3u8', 'video_profile': '360p'},
>         {'id': 'LD', 'container': 'm3u8', 'video_profile': '210p'},
>     ]
>     '''
>     supported_stream_types = [ 'high', 'standard']
>     stream_to_bid = {  '4k': 10, 'fullhd' : 5, 'suprt-high' : 4, 'super' : 3, 'high' : 2, 'standard' :1, 'topspeed' :96}
>     '''
>     ids = ['4k','BD', 'TD', 'HD', 'SD', 'LD']
>     vd_2_id = {10: '4k', 19: '4k', 5:'BD', 18: 'BD', 21: 'HD_H265', 2: 'HD', 4: 'TD', 17: 'TD_H265', 96: 'LD', 1: 'SD', 14: 'TD'}
>     id_2_profile = {'4k':'4k', 'BD': '1080p','TD': '720p', 'HD': '540p', 'SD': '360p', 'LD': '210p', 'HD_H265': '540p H265', 'TD_H265': '720p H265'}
> ```
> (line 93 - 114)

> A recent command record:
>
> ```ShellSession
> > you-get ""
>
> [DEBUG] get_content: https://www.google.com/search?tbm=vid&q=
> you-get: version 0.4.1456, a tiny downloader that scrapes the web.
> you-get: Namespace(URL=[''], auto_rename=False, cookies=None, debug=True, extractor_proxy=None, force=False, format=None, help=False, http_proxy=None, info=False, input_file=None, insecure=True, itag=None, json=False, no_caption=True, no_merge=False, no_proxy=False, output_dir='.', output_filename=None, password=None, player=None, playlist=False, skip_existing_file_size_check=False, socks_proxy=None, stream=None, timeout=600, url=False, version=False)
>
> ...
> ```
>
> ```ShellSession
> > you-get -i "http://www.iqiyi.com/v_19rrldw9sg.html"
>
> [DEBUG] get_response: http://www.iqiyi.com/v_19rrldw9sg.html
> [DEBUG] get_content: http://pcw-api.iqiyi.com/video/video/playervideoinfo?tvid=493448700
> [DEBUG] get_content: http://cache.m.iqiyi.com/tmts/493448700/493448700/?t=1602496395115&sc=17d9b3d19c9ebc014c75b567306c3771&src=76f90cbd92f94a2e925d83e8ccd22cb7
> site:                爱奇艺 (Iqiyi)
> title:               狭路第1集
> streams:             # Available quality and codecs
>     [ DEFAULT ] _________________________________
>     - format:        TD
>       container:     m3u8
>       video-profile: 720p
>       m3u8_url:      http://cache.m.iqiyi.com/mus/204067501/b2cb7117e11a2939414d2a6ec56bc7cd/afbe8fd3d73448c9/0/20200530/e6/21/6dc0139ffeee1fab427a004278f9ccda.m3u8?qd_originate=tmts_py&tvid=493448700&bossStatus=0&qd_vip=0&px=&src=3_31_312&prv=&previewType=&previewTime=&from=&qd_time=1602496395659&qd_p=df6860ad&qd_asc=4509687e8893f5a1477ba8f6a5622507&qypid=493448700_04022000001000000000_4&qd_k=17d9b3d19c9ebc014c75b567306c3771&isdol=0&code=2&ff=f4v&iswb=0&qd_s=otv&vf=080f50017317b5435182f75d860bd8a3&np_tag=nginx_part_tag
>     # download-with: you-get --format=TD [URL]
>
>     - format:        TD_H265
>       container:     m3u8
>       video-profile: 720p H265
>       m3u8_url:      http://cache.m.iqiyi.com/mus/204067501/f7e5bcb81b57bb518afe32756b472c32/afbe8fd3d73448c9/0/20200530/e6/21/116a394c364344998ed30938b285f76e.m3u8?qd_originate=tmts_py&tvid=493448700&bossStatus=0&qd_vip=0&px=&src=3_31_312&prv=&previewType=&previewTime=&from=&qd_time=1602496395660&qd_p=df6860ad&qd_asc=91c7f88bbd35ff3c8e4ff1bafe55712f&qypid=493448700_04022000001000000000_17&qd_k=17d9b3d19c9ebc014c75b567306c3771&isdol=0&code=1&ff=265ts&iswb=0&qd_s=otv&vf=8f8a38015036f422aa131e0b3b6f4150&np_tag=nginx_part_tag
>     # download-with: you-get --format=TD_H265 [URL]
>
>     - format:        HD
>       container:     m3u8
>       video-profile: 540p
>       m3u8_url:      http://cache.m.iqiyi.com/mus/204067501/a0326d0224db24415f2c0ccacc800266/afbe8fd3d73448c9/0/20200722/99/74/ac79dbaf2d29c339446c20dc8f119baf.m3u8?qd_originate=tmts_py&tvid=493448700&bossStatus=0&qd_vip=0&px=&src=3_31_312&prv=&previewType=&previewTime=&from=&qd_time=1602496395660&qd_p=df6860ad&qd_asc=ed4cd3311bfe91ab7437acb0fe649848&qypid=493448700_04022000001000000000_2&qd_k=17d9b3d19c9ebc014c75b567306c3771&isdol=0&code=2&ff=f4v&iswb=0&qd_s=otv&vf=1e4a86e682fff6e8b65e14069df7730d&np_tag=nginx_part_tag
>     # download-with: you-get --format=HD [URL]
>
>     - format:        HD_H265
>       container:     m3u8
>       video-profile: 540p H265
>       m3u8_url:      http://cache.m.iqiyi.com/mus/204067501/7c6dbafe562344ebc61ff07b135f9488/afbe8fd3d73448c9/0/20200722/99/74/5ee591986d973066af71413f0af611d0.m3u8?qd_originate=tmts_py&tvid=493448700&bossStatus=0&qd_vip=0&px=&src=3_31_312&prv=&previewType=&previewTime=&from=&qd_time=1602496395660&qd_p=df6860ad&qd_asc=5057dc1e3f16a22be1cceb9bfc1259b1&qypid=493448700_04022000001000000000_21&qd_k=17d9b3d19c9ebc014c75b567306c3771&isdol=0&code=1&ff=265ts&iswb=0&qd_s=otv&vf=4915a03ba53e80d8033ba65fc856a4c5&np_tag=nginx_part_tag
>     # download-with: you-get --format=HD_H265 [URL]
>
>     - format:        SD
>       container:     m3u8
>       video-profile: 360p
>       m3u8_url:      http://cache.m.iqiyi.com/mus/204067501/1eb7ad368d09ba5eb9a3f678cfcdbfeb/afbe8fd3d73448c9/0/20200602/e6/21/b0a3195b356a917cb076cb2214c88cf2.m3u8?qd_originate=tmts_py&tvid=493448700&bossStatus=0&qd_vip=0&px=&src=3_31_312&prv=&previewType=&previewTime=&from=&qd_time=1602496395660&qd_p=df6860ad&qd_asc=8213d1a271239ec53659f17e67074099&qypid=493448700_04022000001000000000_1&qd_k=17d9b3d19c9ebc014c75b567306c3771&isdol=0&code=2&ff=f4v&iswb=0&qd_s=otv&vf=49fa1a891ae2b6ce35d65c09eb27593b&np_tag=nginx_part_tag
>     # download-with: you-get --format=SD [URL]
>
>     - format:        LD
>       container:     m3u8
>       video-profile: 210p
>       m3u8_url:      http://cache.m.iqiyi.com/mus/204067501/f0e43bb3eebf247eb9c263c751d85328/afbe8fd3d73448c9/0/20190924/2f/5d/cf6fcc6ac4680d968a2cb4cca42bb9ea.m3u8?qd_originate=tmts_py&tvid=493448700&bossStatus=0&qd_vip=0&px=&src=3_31_312&prv=&previewType=&previewTime=&from=&qd_time=1602496395659&qd_p=df6860ad&qd_asc=de0371da358f56e799f213d36405ae5a&qypid=493448700_04022000001000000000_96&qd_k=17d9b3d19c9ebc014c75b567306c3771&isdol=0&code=2&ff=f4v&iswb=0&qd_s=otv&vf=58912349bed09e951c50f23a87ba4322&np_tag=nginx_part_tag
>     # download-with: you-get --format=LD [URL]
> ```

Where had the higher quality versions gone?.. I wonder:

If a VIP cookie were loaded in a [similar manner](https://github.com/soimort/you-get/pull/2803): would it help with the situation? (didn't for Youku, in my previous test)




<br/><br/>

## Off-Topic

My apologies for making a random pull request...

Didn't manage to figure out how properly, and it gets locked... so quickly, that no further remediation could be done.


<br/>

If there must be at least one commit... that has to take place on a repository file:

That thing should go to the "/README.md", perhaps, rather than remaining locked needlessly...

**Pull request merged**